src/irmin-pack/unix/file_manager.ml

```diff
@@ -457,13 +462,16 @@ struct
       (* [overwrite] is ignored for index *)
       Index.v ~fresh:true ~flush_callback ~readonly ~throttle ~log_size root
     in
+    let make_lower () = Ok None in
```

Comment:
:+1: I like how you mock functions and then in a later commit fill in the implementation

Reply:
On the other hand, maybe this commit could be squashed with 2996200a4a, which is relatively small. That would remove the little bit of time in Git history where this functionality is mocked.

Reply:
These commits walk through how I thought about building the implementation, but in hindsight I think these two commits can be combined. :+1:

Reply:
Mm, actually now I see more why it is the way it is. I can't squash these two commits (in either direction) if I want a separate commit for the `add_volume` API (which I do) -- add_volume depends on a lower being present and creating an empty volume depends on being able to add a volume. I could squash commit 2 and 3 together, but I'd rather not put "adding the add_volume API" and "creating an empty volume on store create" into one commit since they are nicely self-contained/described changes in my mind.

Reply:
Aha, ok. Then fine to keep as is. Thanks for the elaboration!